modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsWriter.java

```diff
@@ -0,0 +1,580 @@
+                buf.putLong(physicalReads);
+            });
+    }
+
+    /**
+     * @param sesId Session id.
+     * @param taskName Task name.
+     * @param startTime Start time in milliseconds.
+     * @param duration Duration.
+     * @param affPartId Affinity partition id.
+     */
+    public void task(IgniteUuid sesId, String taskName, long startTime, long duration, int affPartId) {
+        FileWriter writer = fileWriter;
+
+        if (writer == null)
+            return;
+
+        boolean needWriteStr = !writer.stringCached(taskName);
+        byte[] strBytes = needWriteStr ? taskName.getBytes() : null;
+
+        doWrite(OperationType.TASK, () -> {
+            int size = 24 + 1 + 4 + 8 + 8 + 4;
+
+            if (needWriteStr)
+                size += 4 + strBytes.length;
+
+            return size;
+        }, buf -> {
+            writeIgniteUuid(buf, sesId);
+            buf.put(needWriteStr ? (byte)1 : 0);
+            buf.putInt(taskName.hashCode());
+
+            if (needWriteStr) {
+                buf.putInt(strBytes.length);
+                buf.put(strBytes);
+            }
+
+            buf.putLong(startTime);
+            buf.putLong(duration);
+            buf.putInt(affPartId);
+        });
+    }
+
+    /**
+     * @param sesId Session id.
+     * @param queuedTime Time job spent on waiting queue.
+     * @param startTime Start time in milliseconds.
+     * @param duration Job execution time.
+     * @param timedOut {@code True} if job is timed out.
+     */
+    public void job(IgniteUuid sesId, long queuedTime, long startTime, long duration, boolean timedOut) {
+        doWrite(OperationType.JOB,
+            () -> 24 + 8 + 8 + 8 + 1,
+            buf -> {
+                writeIgniteUuid(buf, sesId);
+                buf.putLong(queuedTime);
+                buf.putLong(startTime);
+                buf.putLong(duration);
+                buf.put(timedOut ? (byte)1 : 0);
+            });
+    }
+
+    /**
+     * @param op Operation type.
+     * @param sizeSupplier Record size supplier.
+     * @param writer Record writer.
+     */
+    private void doWrite(OperationType op, IntSupplier sizeSupplier, Consumer<ByteBuffer> writer) {
+        FileWriter fileWriter = this.fileWriter;
+
+        // Writer stopping.
+        if (fileWriter == null)
+            return;
+
+        int size = sizeSupplier.getAsInt();
+
+        SegmentedRingByteBuffer.WriteSegment seg = reserveBuffer(fileWriter, op, size);
+
+        if (seg == null)
+            return;
+
+        writer.accept(seg.buffer());
+
+        seg.release();
+    }
+
+    /**
+     * Reserves buffer's write segment.
+     *
+     * @param fileWriter File writer.
+     * @param type Operation type.
+     * @param size Record size.
+     * @return Buffer's write segment or {@code null} if not enought space or writer stopping.
+     */
+    private SegmentedRingByteBuffer.WriteSegment reserveBuffer(FileWriter fileWriter, OperationType type, int size) {
```

Comment:
this method used only once. Let's inline it.

Reply:
Done.